### PR TITLE
WGSL textureNumLevels execution tests

### PR DESCRIPTION
**Pull request by greggman**
<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/format_info.ts [hunk 1]
```diff
@@ -1554,6 +1554,26 @@ export const kValidTextureFormatsForCopyE2T = [
   'rgba32float',
 ] as const;
 
+/** List of storage texture formats */
```
Comment (alan-baker) [inline src/webgpu/format_info.ts:1557]: Unused variable.

File: src/webgpu/shader/execution/expression/call/builtin/textureNumLevels.spec.ts [hunk 2]
```diff
@@ -34,12 +63,60 @@ Parameters
...
+  .fn(t => {
+    const { texture_type, sampled_type, view_type } = t.params;
+    const { format } = kSampleTypeInfo[sampled_type];
+
+    const viewDimension = kTextureTypeToViewDimension[texture_type];
+    const dimension = getTextureDimensionFromView(viewDimension);
+    const isCube = texture_type.includes('cube');
+    const width = 64;
+    const height = dimension === '1d' ? 1 : width;
+    const depthOrArrayLayers = isCube ? 6 : 1;
```
Comment (alan-baker) [inline src/webgpu/shader/execution/expression/call/builtin/textureNumLevels.spec.ts:86]: Isn't a depth of 1 problematic for mip count > 1 on 3D textures?

File: src/webgpu/format_info.ts (hunk 1 above)
Comment (greggman) [reply src/webgpu/format_info.ts:1557]: thanks! removed.

File: src/webgpu/shader/execution/expression/call/builtin/textureNumLevels.spec.ts (hunk 2 above)
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/textureNumLevels.spec.ts:86]: a 3d texture can have depthOrArrayLayers = 1 and mipLevelCount = 1 + floor(log2(max(width, height, depthOrArrayLayers))). width is 64 so it has mip levels

File: src/webgpu/format_info.ts (hunk 1 above)
Comment (alan-baker) [reply src/webgpu/format_info.ts:1557]: I think you forgot to push this change.

Comment (greggman) [reply src/webgpu/format_info.ts]: I just noticed that too. Pushed.

**Review by alan-baker — APPROVED** [review]

**Pull request merged**